mkversion.sh

```diff
@@ -32,6 +32,12 @@ if [ ! -z "$1" ]; then
     o=shell
 fi
 
+# If the version is passed in the environment during build, let's use that
```

Comment:
Let's please fix any such broken tarballs now, while it's early and we haven't suffered any major pain due to that.